rust/Cargo.lock

```diff
@@ -0,0 +1,21 @@
+[root]
+name = "suricata"
+version = "4.0.0-dev"
```

Comment:
Cargo.lock is actually a generated file. Its generated from Cargo.toml. I'm not sure exactly when it gets updated tho. Will have to look into this more closely.

The lock file allows us to pin the dependency versions for reproducible builds, so I think its worthwhile using. There is some argument on whether or not it should even go into version control.  But for an end-user app, as opposed to a library it seems to make sense to check it in. That way everyone is building against the same dependencies.